library/src/main/java/com/schibsted/spain/barista/BaristaRecyclerViewActions.java

```diff
@@ -24,21 +26,21 @@ public static void clickRecyclerViewItem(@IdRes int recyclerViewId, int... posit
   }
 
   private static void performClick(@IdRes int recyclerViewId, int position) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
         RecyclerViewActions.actionOnItemAtPosition(position, clickUsingPerformClick()));
   }
 
   public static void scrollTo(int recyclerViewId, int position) {
-    onView(withId(recyclerViewId)).perform(scrollToPosition(position));
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(scrollToPosition(position));
   }
 
   public static void clickRecyclerViewItemChild(@IdRes int recyclerViewId, int position, @IdRes int itemToClickId) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
         RecyclerViewActions.actionOnItemAtPosition(position, clickChildWithId(itemToClickId)));
   }
 
   public static void clickRecyclerViewItemChild(@IdRes int recyclerViewId, int position, String text) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
```

Comment:
no no it's not your english.. I just needed more context :p 
now it's much better, thanks!